### PR TITLE
ddl: wait until there is an owner in the cluster

### PR DESCRIPTION
**Pull request by xiongjiwei**
Signed-off-by: xiongjiwei <xiongjiwei1996@outlook.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/36659

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/ddl.go [hunk 1]
```diff
@@ -681,6 +685,24 @@ func (d *ddl) Start(ctxPool *pools.ResourcePool) error {
 	return nil
 }
 
+// we have to wait at least an owner in the cluster.
```
Comment (tangenta) [inline ddl/ddl.go:688]:
```suggestion
// waitOwner is used to wait the DDL owner to be elected in the cluster.
```

File: ddl/ddl.go [hunk 2]
```diff
@@ -681,6 +685,24 @@ func (d *ddl) Start(ctxPool *pools.ResourcePool) error {
...
+func (d *ddl) waitOwner(ctx context.Context) error {
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			_, err := d.ownerManager.GetOwnerID(ctx)
```
Comment (tangenta) [inline ddl/ddl.go:697]: Can we use etcd client's `Watch()` instead? Thus, the timer can be removed.

**Review by tangenta — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**